launch_xml/README.md

```diff
@@ -0,0 +1,92 @@
+# launch_xml
+
+This package provides an abstraction of the XML tree.
+
+## XML front-end mapping rules
+
+### Accessing xml attributes
+
+When having an xml tag like:
+
+```xml
+<tag attr='2'/>
+```
+
+If the entity `e` is wrapping it, the following two statements would be true:
+```python
+hasattr(e, 'attr') == True
+e.attr == '2'
+```
+
+As a general rule, the value of the attribute is returned as an string.
+Conversion to `float` or `int` should be explicitly done in the parser method, or in a substitution.
+For handling lists, see `Built-in Substitutions` section.
+
+### Accessing XML children that aren't an action or a substitution:
+
+In this xml:
+
+```xml
+<executable cmd="ls">
+    <env name="a" value="100"/>
+    <env name="b" value="stuff"/>
+</node>
+```
+
+The `env` childs could be accessed like:
+
+```python
+len(e.env) == 2
+e.env[0].name == 'a'
+e.env[0].value == '100'
+e.env[1].name == 'b'
+e.env[1].value == 'stuff'
+```
+
+In these cases, `e.env` is a list of entity, which could be accessed in the same abstract way.
+
+### Accessing XML children that are an action or a substitution:
+
+The entity will keep track of which `children` where accessed or not.
```

Comment:
@ivanpauno 
```suggestion
The entity will keep track of which `children` were accessed or not.
```

Reply:
I'm also curious, about this functionality, under what circumstances do you think this would be needed? I'm almost sure we don't  have one of those in the current XML format document.